### PR TITLE
Remove zapup library.

### PR DESCRIPTION
**Pull request by Gerrit91**
```BREAKING_CHANGE
The `OIDCFlow` function of this library now requires a logger. When you use this function, please pass a valid logger.
```

### CODE REVIEW

File: rest/middleware.go [hunk 1]
```diff
@@ -39,78 +46,105 @@ func (w *loggingResponseWriter) Content() string {
...
-			zap.String("method", rq.Method),
-			zap.String("uri", rq.URL.RequestURI()),
-			zap.String("route", req.SelectedRoutePath()),
+
+		clone := *logger
+		requestLogger := &clone
+
+		fields := []any{
+			"rqid", requestID,
+			"remoteaddr", strings.Split(rq.RemoteAddr, ":")[0],
```
Comment (majst01) [inline rest/middleware.go:67]: why not simply remoteAddr ?

File: rest/middleware.go [hunk 2]
```diff
@@ -39,78 +46,105 @@ func (w *loggingResponseWriter) Content() string {
...
 		}
-		fields := []zap.Field{
-			rqid,
-			zap.String("remoteaddr", strings.Split(rq.RemoteAddr, ":")[0]),
-			zap.String("method", rq.Method),
-			zap.String("uri", rq.URL.RequestURI()),
-			zap.String("route", req.SelectedRoutePath()),
+
+		clone := *logger
+		requestLogger := &clone
```
Comment (majst01) [inline rest/middleware.go:63]: please remove, this is a massive overhead, for separating disjunctive requests, requestID is added. A Benchmark would help to see the impact.

**Comment by majst01** [comment]
Lets check again if we can reuse some existing logging extensions to zap which does this already.

File: rest/middleware.go (hunk 2 above)
Comment (Gerrit91) [reply rest/middleware.go]: This was really unnecessary because `.With` already creates a clone (see source code).

File: rest/middleware.go (hunk 1 above)
Comment (Gerrit91) [reply rest/middleware.go]: Done.

File: rest/middleware.go [hunk 3]
```diff
@@ -39,78 +45,105 @@ func (w *loggingResponseWriter) Content() string {
...
+				log.Errorw("cannot get user from request", "error", err, "got", hmerr.Got, "want", hmerr.Want)
 			} else {
-				log.Error("cannot get user from request", zap.Error(err))
+				log.Errorw("cannot get user from request", "error", err)
 			}
+
 			err = resp.WriteHeaderAndEntity(http.StatusForbidden, httperrors.NewHTTPError(http.StatusForbidden, err))
 			if err != nil {
-				log.Error("writeHeaderAndEntity", zap.Error(err))
+				log.Error("error sending response", "error", err)
```
Comment (majst01) [inline rest/middleware.go:122]: Errorw

**Review by majst01 — CHANGES_REQUESTED** [review]
should be checked with other projects using this library as well.

**Review by majst01 — APPROVED** [review]

**Pull request merged**